default/scripting/buildings/XENORESURRECTION_LAB.focs.txt

```diff
@@ -13,6 +13,26 @@ BuildingType
         ]
     ]
     EnqueueLocation = [[ENQUEUE_BUILD_ONE_PER_PLANET]]
+    effectsgroups = [
+        EffectsGroup
+            scope = And [
+                Source
+                ContainedBy HasSpecial name = "EXTINCT_BANFORO_SPECIAL"
```

Comment:
Perhaps better to put the ContainedBy condition in the activation condition, so the scope is just Source?
Also, maybe make it

    And [
        Object id = Source.PlanetID
        HasSpecial name = "..."
    ]

so it's not checking any other objects for the presence of the special. The ContainedBy condition doesn't presently limit the containers it searches very well... and isn't necessary in a case like this where you only want the container of a single directly-referencable object (the source), and not anything that contains and object matching a more complicated condition.